dev.py

```diff
@@ -486,6 +488,24 @@ def setup_build(cls, dirs, args):
                 return
         if args.werror:
             cmd += ["--werror"]
+        if args.debug or args.release:
+            if args.debug and args.release:
+                raise ValueError("Set at most one of `--debug` and `--release`!")
+            if args.debug:
+                cmd += ["-Dbuildtype=debug"]
+                cflags_unwanted = ('-O1', '-O2', '-O3')
+            elif args.release:
+                cmd += ["-Dbuildtype=release"]
+                cflags_unwanted = ('-O0', '-O1', '-O2')
+            if 'CFLAGS' in os.environ.keys():
+                # Check that CFLAGS doesn't contain something that supercedes -O0
+                # for a plain debug build (conda envs tend to set -O2)
+                cflags = os.environ['CFLAGS'].split()
+                for flag in cflags_unwanted:
+                    if flag in cflags:
+                        raise ValueError(f"A plain debug build isn't possible, "
+                                         f"because CFLAGS contains `{flag}`."
+                                          "Please also check CXXFLAGS and FFLAGS.")
```

Comment:
Since you defined `cflags_unwanted` for both build types above, maybe we should generalize the error message?

`CFLAGS="-O0" python dev.py build --release`

```
ValueError: A plain debug build isn't possible, because CFLAGS contains `-O0`.Please also check CXXFLAGS and FFLAGS.
```

Maybe we could string sub on the build type?

```diff
--- a/dev.py
+++ b/dev.py
@@ -492,9 +492,11 @@ class Build(Task):
             if args.debug and args.release:
                 raise ValueError("Set at most one of `--debug` and `--release`!")
             if args.debug:
+                b_str = "debug"
                 cmd += ["-Dbuildtype=debug"]
                 cflags_unwanted = ('-O1', '-O2', '-O3')
             elif args.release:
+                b_str = "release"
                 cmd += ["-Dbuildtype=release"]
                 cflags_unwanted = ('-O0', '-O1', '-O2')
             if 'CFLAGS' in os.environ.keys():
@@ -503,7 +505,7 @@ class Build(Task):
                 cflags = os.environ['CFLAGS'].split()
                 for flag in cflags_unwanted:
                     if flag in cflags:
-                        raise ValueError(f"A plain debug build isn't possible, "
+                        raise ValueError(f"A plain {b_str} build isn't possible, "
                                          f"because CFLAGS contains `{flag}`."
                                           "Please also check CXXFLAGS and FFLAGS.")
```

``

Reply:
Thanks, good point. Addressed now.